torchaudio_contrib/layers.py

```diff
@@ -218,38 +218,43 @@ class TimeStretch(_ModuleNoStateBuffers):
 
     Args:
 
-        rate (float): rate to speed up or slow down by. 
+        fixed_rate (float): rate to speed up or slow down by.
             Defaults to 1 (do nothing).
         hop_length (int): Number audio of frames between STFT columns.
             Defaults to 512.
         num_freqs (int, optional): number of filter banks from stft.
             Defaults to 1025.
     """
 
-    def __init__(self, rate=1., hop_length=512, num_freqs=1025):
+    def __init__(self, fixed_rate=1., hop_length=512, num_freqs=1025):
```

Comment:
perhaps we shouldn't set default values for any of these? unlike librosa we don't have any default `n_fft` or `hop_length` over the framework. `rate` would be also raising error when user have no idea about it.

Reply:
Yeah definitely agree on the `fixed_rate` and `num_freqs`. Since `torch.stft` does handle the `hop_length` not being passed, should we do it also in the same way for `TimeStretch`? As in something like:

```python
class TimeStretch(_ModuleNoStateBuffers):
    """
    Stretch stft in time without modifying pitch for a given rate.

    Args:

        fixed_rate (float): rate to speed up or slow down by.
        num_freqs (int, optional): number of filter banks from stft.
        hop_length (int): Number audio of frames between STFT columns.
            Defaults to fft_length//4.
            """

    def __init__(self, fixed_rate, num_freqs, hop_length=None):
        super(TimeStretch, self).__init__()
        if hop_length is None:
            hop_length = int(2*(num_freqs-1)/4)
        ...
```

Reply:
Hm, I'd prefer to be on the even safer side and keep `hop_length` as a required argument.